src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -1653,6 +1660,34 @@ def get_finetune_lr_dict(self, lr: float) -> Dict[str, float]:
             if not self.ddp_silent_mode:
                 self.sg_logger.close()
 
+    def _maybe_eval_before_resume(self, context, silent_mode):
+        if self._eval_before_resume:
+            if not silent_mode:
+                logger.info(f"Couldn't fetch {self.metric_to_watch} from the checkpoint.\n Running test on the " f"validation data before resuming training...")
+            if self.ema:
+                keep_model = self.net
+                self.net = self.ema_model.ema
+
+            self.net.eval()
+            self._reset_metrics()
+            self.valid_metrics.to(device_config.device)
+            valid_metrics_dict = self.evaluate(
+                data_loader=self.valid_loader,
+                metrics=self.valid_metrics,
+                evaluation_type=EvaluationType.TEST,
+                silent_mode=silent_mode,
+            )
+            self._save_checkpoint(
```

Comment:
Why _save_checkpoint here? We should just obtain the metrics. If the goal is to overwrite the previous best checkpoint with a new metrics - I think this is very unexpected behavior and I personally against such move.

Reply:
It is necessary to save the checkpoint with these metrics, as continuing the training might have inferior results, for example:
- epoch 0, completed with XXX accuracy score.
Resuming training, and we change metric_to_watch to top5
- We evaluate top5, we get 0.3, **dont save checkpoint**  like you suggested.
- Then in the end of epoch 1, we get 0.2, then it would be considered best and we would save the non optimal checkpoint as best.

Why is it unexpected behaviour?
The checkpoint_best.pth should always hold the best checkpoint according to the metric_to_watch at that time.